File: dotnet/weak_cryptography/Encryption.cs
using System.Security.Cryptography;
using System.Text;

public class Encryption
{
    // VULN 1: DES cipher used to encrypt stored credit card data
    public byte[] EncryptCardData(string cardNumber, byte[] key)
    {
        using var des = new DESCryptoServiceProvider();
        des.Key = key;
        des.GenerateIV();
        using var encryptor = des.CreateEncryptor();
        byte[] inputBytes = Encoding.UTF8.GetBytes(cardNumber);
        return encryptor.TransformFinalBlock(inputBytes, 0, inputBytes.Length);
    }

    // VULN 2: RC2 cipher used to encrypt user session data stored server-side
    public byte[] EncryptSessionData(string sessionPayload, byte[] key)
    {
        using var rc2 = new RC2CryptoServiceProvider();
        rc2.Key = key;
        rc2.GenerateIV();
        using var encryptor = rc2.CreateEncryptor();
        byte[] inputBytes = Encoding.UTF8.GetBytes(sessionPayload);
        return encryptor.TransformFinalBlock(inputBytes, 0, inputBytes.Length);
    }

    // VULN 3: AES in ECB mode with no IV - reveals patterns in encrypted order data
    public byte[] EncryptOrderData(string orderJson, byte[] key)
    {
        using var aes = Aes.Create();
        aes.Key = key;
        aes.Mode = CipherMode.ECB;
        aes.Padding = PaddingMode.PKCS7;
        using var encryptor = aes.CreateEncryptor();
        byte[] inputBytes = Encoding.UTF8.GetBytes(orderJson);
        return encryptor.TransformFinalBlock(inputBytes, 0, inputBytes.Length);
    }
}

File: python/weak_cryptography/encryption.py
from Crypto.Cipher import DES, ARC4
from Crypto.Cipher import AES
import os


# VULN 1: DES encryption for storing payment card data (key too short, broken cipher)
def encrypt_card_number(card_number):
    key = b"shopkey"[:8]  # DES requires 8-byte key
    cipher = DES.new(key, DES.MODE_ECB)
    padded = card_number.ljust(16).encode()
    return cipher.encrypt(padded)


# VULN 2: RC4 (ARC4) stream cipher - cryptographically broken
def encrypt_session_data(data, key=b"session_rc4_key!"):
    cipher = ARC4.new(key)
    return cipher.encrypt(data.encode())


# VULN 3: AES-ECB mode - does not provide semantic security (reveals patterns)
def encrypt_user_pii(data):
    key = b"shopist16bytekey"
    cipher = AES.new(key, AES.MODE_ECB)
    padded = data.ljust(32).encode()
    return cipher.encrypt(padded)
